runtime/testy/runtime.py

```diff
@@ -64,33 +66,46 @@ def runtime():
     print("Funtime Runtime Had Too Much Fun")
     print(traceback.print_exception(*sys.exc_info()))
 
-def runStudentCode(badThingsQueue, stateQueue, pipe):
+def runStudentCode(badThingsQueue, stateQueue, pipe, testName = "", maxIter = 0):
   try:
     import signal
-    def timed_out_handler(signum, frame):
+    def timedOutHandler(signum, frame):
       raise TimeoutError("studentCode timed out")
-    signal.signal(signal.SIGALRM, timed_out_handler)
+    signal.signal(signal.SIGALRM, timedOutHandler)
+
+    def checkTimedOut(func, *args):
```

Comment:
I think this function is misnamed. It doesn't really check for timeout at all, it just schedules an alarm and then runs something.